core/common/lib/transform-lib/src/main/java/org/eclipse/edc/transform/transformer/edc/from/JsonObjectFromDataAddressTransformer.java

```diff
@@ -45,7 +49,14 @@ public JsonObjectFromDataAddressTransformer(JsonBuilderFactory jsonBuilderFactor
 
         builder.add(TYPE, EDC_NAMESPACE + "DataAddress");
 
-        transformProperties(dataAddress.getProperties(), builder, typeManager.getMapper(typeContext), context);
+        BiFunction<String, Object, JsonValue> func = (k, v) -> {
+            if (k.equals(EDC_DATA_ADDRESS_RESPONSE_CHANNEL)) {
+                return context.transform(v, JsonObject.class);
+            }
+            return typeManager.getMapper(typeContext).convertValue(v, JsonValue.class);
```

Comment:
Can you resolve the mapper just once and use the closure reference (see line 48 above)?